api/tests/identity-access-management/unit/domain/models/ClaimsManager.test.js

```diff
@@ -0,0 +1,152 @@
+
+      // when
+      const claimManager = new ClaimManager({ claimMapping, additionalClaims });
+      const result = claimManager.pickAdditionalClaims({ given_name: 'Bob', foo: 'Uncle' });
+
+      // then
+      expect(result).to.deep.equal({ foo: 'Uncle' });
+    });
+
+    it('returns an empty object when user info are null', async function () {
+      // when
+      const claimManager = new ClaimManager();
+      const result = claimManager.pickAdditionalClaims(null);
+
+      // then
+      expect(result).to.deep.equal({});
+    });
+  });
+
+  describe('#hasMissingClaims', function () {
+    it('returns true when some claims are missing', async function () {
+      // given
+      const claimMapping = { firstName: ['foo'], lastName: ['bar'] };
+
+      // when
+      const claimManager = new ClaimManager({ claimMapping });
+      const hasMissingClaims = claimManager.hasMissingClaims({ foo: 'foofoo' });
+
+      // then
+      expect(hasMissingClaims).to.equal(true);
+    });
+
+    it('returns true when some additional claims are missing', async function () {
+      // given
+      const claimMapping = { firstName: ['foo'], lastName: ['bar'] };
+
+      // when
+      const claimManager = new ClaimManager({ claimMapping, additionalClaims: ['baz'] });
+      const hasMissingClaims = claimManager.hasMissingClaims({ foo: 'foofoo', bar: 'barbar' });
+
+      // then
+      expect(hasMissingClaims).to.equal(true);
+    });
+
+    it('returns false when all claims are present', async function () {
+      // given
+      const claimMapping = { firstName: ['foo'], lastName: ['bar'] };
+
+      // when
+      const claimManager = new ClaimManager({ claimMapping });
+      const hasMissingClaims = claimManager.hasMissingClaims({ foo: 'foofoo', bar: 'barbar' });
+
+      // then
+      expect(hasMissingClaims).to.equal(false);
+    });
+  });
+
+  describe('#getMissingClaims', function () {
+    it('returns the missing claims', async function () {
+      // given
+      const claimMapping = { firstName: ['foo'], lastName: ['bar'], id: ['sub'] };
+
+      // when
+      const claimManager = new ClaimManager({ claimMapping, additionalClaims: ['baz', 'fox'] });
+      const result = claimManager.getMissingClaims({ sub: '', foo: 'foofoo', baz: '' });
+
+      // then
+      expect(result).to.deep.equal(['bar', 'sub', 'baz', 'fox']);
+    });
+
+    it('returns an empty array when all claims are present', async function () {
+      // given
+      const claimMapping = { firstName: ['foo'], lastName: ['bar'] };
+
+      // when
+      const claimManager = new ClaimManager({ claimMapping });
+      const result = claimManager.getMissingClaims({ foo: 'foofoo', bar: 'barbar' });
+
+      // then
+      expect(result).to.deep.equal([]);
+    });
+  });
+
+  describe('#hasAdditionalClaims', function () {
+    it('returns true if has additional claims', async function () {
+      // given
+      const additionalClaims = ['baz'];
```

Comment:
Ce serait plus parlant si ce test avait plusieurs additionalClaims et avec des noms faisant penser qu'ils sont attendus/nécessaires, par exemple (et adapter le test en conséquence) :

```suggestion
      const additionalClaims = ['aWantedClaim', 'anotherWantedClaim'];
```